controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -201,8 +201,8 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 		if sc.Spec.MultiCloudGateway.ExternalPgConfig != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName != "" {
 			nb.Spec.ExternalPgSecret = &corev1.SecretReference{Name: sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName, Namespace: sc.Namespace}
 			nb.Spec.ExternalPgSSLUnauthorized = sc.Spec.MultiCloudGateway.ExternalPgConfig.AllowSelfSignedCerts
-			if sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretName != "" {
-				nb.Spec.ExternalPgSSLRequired = true
+			nb.Spec.ExternalPgSSLRequired = sc.Spec.MultiCloudGateway.ExternalPgConfig.EnableTLS
+			if sc.Spec.MultiCloudGateway.ExternalPgConfig.EnableTLS && sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretName != "" {
 				nb.Spec.ExternalPgSSLSecret = &corev1.SecretReference{Name: sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretName, Namespace: sc.Namespace}
 			}
```

Comment:
We need to log error when validation fails or we won't know what went wrong. Even better to return an invalid config error.

Reply:
Its not an invalid config if it fails. It can still work with just having the TLS enabled and not have any client side authetication.

Reply:
But, if EnableTLS is false and TLSSecretName is provided, then it is an invalid config right? Shouldn't we let the users know about it?

Reply:
It will be blocked in the UI itself. Only when the checkbox to "Enable TLS/SSL" is selected, then only users are allowed to pass the optional certs. So it will always be true.

Unless we want to add it for CLI purposes, in which case, this should be part of the webhook validations

Reply:
We can't depend on UI for validation and webhooks aren't available in ocs-operator.
An error log is the only option we have along with updates to status.conditions to explain what's wrong.

Reply:
ok done with the changes then